app/src/main/java/com/bc/gordiansigner/service/ContactService.kt

```diff
@@ -0,0 +1,69 @@
+package com.bc.gordiansigner.service
+
+import com.bc.gordiansigner.helper.ext.fromJson
+import com.bc.gordiansigner.helper.ext.newGsonInstance
+import com.bc.gordiansigner.model.KeyInfo
+import com.bc.gordiansigner.service.storage.file.FileStorageApi
+import com.bc.gordiansigner.service.storage.file.rxCompletable
+import com.bc.gordiansigner.service.storage.file.rxSingle
+import com.bc.gordiansigner.service.storage.sharedpref.SharedPrefApi
+import io.reactivex.Completable
+import io.reactivex.Single
+import javax.inject.Inject
+
+class ContactService @Inject constructor(
```

Comment:
We need to declare this service in `ServiceModule` as well.

Reply:
Oh, I thought binding directly through @Inject in constructor doesn't need to provide using module?

Reply:
@hvthhien Yes but we need to declare these services as singleton. The purpose is something like caching, consistent storage APIs (i.e. using the same `FileStorageApi` instance means using the same key for encrypt/decrypt data)

Reply:
for singleton purpose only, How about adding an annotation @Singleton right before the class declaration? I think using binding will consume fewer resources by generating fewer classes than wrapping them in a module.

Reply:
@hvthhien I think that's fine to reduce the generated classes. I'm using this kind of declaration in several projects to make it more clear for reading code and maintain as well. When we deep dive into the `ServiceModule`, all of providers will be in same scope and even if we need to change it later on, it's easier to change without be aware all providers. I prefer the module-based due to the easy to read, easy to maintain later. 
I think declare the scope annotation on constructor is fine but it would introduce the fragmented code and hard to maintain later. 
You can take a look on another modules like `AppModule`, `RepositoryModule`...